src/ert/gui/model/snapshot.py

```diff
@@ -276,45 +283,55 @@ def _add_snapshot(self, snapshot: EnsembleSnapshot, iter_: str) -> None:
         self.rowsInserted.emit(parent, snapshot_tree.row(), snapshot_tree.row())
 
     @override
-    def columnCount(self, parent: QModelIndex | None = None) -> int:
+    def columnCount(
+        self, parent: QModelIndex | QPersistentModelIndex | None = None
+    ) -> int:
         if parent and isinstance(parent.internalPointer(), RealNode):
             return FM_STEP_COLUMN_SIZE
         return 1
 
-    def rowCount(self, parent: QModelIndex | None = None) -> int:
+    def rowCount(
+        self, parent: QModelIndex | QPersistentModelIndex | None = None
+    ) -> int:
         if parent is None:
             parent = QModelIndex()
         parent_item = self.root if not parent.isValid() else parent.internalPointer()
 
         if parent.column() > 0:
             return 0
 
-        return len(parent_item.children)
+        return len(parent_item.children)  # type: ignore
 
     @overload
-    def parent(self, child: QModelIndex) -> QModelIndex: ...
+    def parent(self) -> QObject: ...
     @overload
-    def parent(self) -> QObject | None: ...
+    def parent(self, child: QModelIndex | QPersistentModelIndex) -> QModelIndex: ...
     @override
-    def parent(self, child: QModelIndex | None = None) -> QObject | None:
+    def parent(
+        self, child: QModelIndex | QPersistentModelIndex | None = None
+    ) -> QObject | QModelIndex:
```

Comment:
Does this also need QObject?
For QModelIndex, is it possible to have the typing include specifically what the index is pointing to?

Reply:
both of the parent overloads come from inherited methods. the override must handle them both